R/slice.R

```diff
@@ -0,0 +1,31 @@
+source("R/connect.R")
+source("R/utils.R")
+
+#' Sends a ["request-slice"](https://github.com/flowr-analysis/flowr/wiki/Interface#the-slice-request) request
+#'
+#' @param con The connection to the server
+#' @param filetoken The filetoken of the file to be sliced
+#' (retrieved by a previous "file-analysis" request)
+#' @param criteria The slice's criteria
+#' @param id The id of the request
+#' @return A list containing the id and the response
+#'
+#' @seealso [initiate_file_analysis()]
+#' @seealso [connect()]
+#'
+#' @export
+request_slice <- function(con, filetoken, criteria, id = get_new_id()) {
+  request <- fromRJSON('{
+    "type":      "request-slice",
+    "id":        "%s",
+    "filetoken": "%s",
+    "criterion": %s
+  }', filetoken, jsonlite::toJSON(criteria))
+  res <- send_request(con, request)
```

Comment:
Hmm this seems redundant to me. The point of `send_request` is that you can just pass a list like I do [in the RStudio addin](https://github.com/flowr-analysis/rstudio-addin-flowr/blob/main/R/slice.R#L34), so converting to JSON, then from JSON, and then (in the `send_request` functino) back to JSON, seems weird. What's the reason you're doing this? Maybe changing the base API makes more sense here, if required

Reply:
I agree with the double-wrapping being redundant. However, the function itself should probably stay, allowing to bridge between the underlying format.

Reply:
I removed the double-wrapping by just passing a list in f1da95c :)

Despite not being used anymore, I also renamed `fromRJSON` to `from_json_string` and documented it.